File: Chapter-2/CalculatorApplication/V1/CalculatorService/CalculatorService.cs
using System;
using System.Collections.Generic;
using System.Fabric;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.ServiceFabric.Services.Communication.Runtime;
using Microsoft.ServiceFabric.Services.Runtime;
using CalculatorService.Interfaces;
using Microsoft.ServiceFabric.Services.Remoting.Runtime;

namespace CalculatorService
{
    /// <summary>
    /// An instance of this class is created for each service instance by the Service Fabric runtime.
    /// </summary>
    internal sealed class CalculatorService : StatelessService, ICalculatorService
    {
        public CalculatorService(StatelessServiceContext context)
            : base(context)
        { }
        protected override IEnumerable<ServiceInstanceListener> CreateServiceInstanceListeners()
        {
            return new[] { new ServiceInstanceListener(context => this.CreateServiceRemotingListener(context)) };
        }
        public Task<int> Add(int a, int b)
        {
            return Task.FromResult<int>(a + b);
        }
        public Task<int> Subtract(int a, int b)
        {
            return Task.FromResult<int>(a - b);
        }
    }
}

File: Chapter-12/CalculatorApplication/Calculator/src/statelessservice/CalculatorService.java
package statelessservice;

import java.util.concurrent.CompletableFuture;
import java.util.List;
import java.util.ArrayList;

import system.fabric.CancellationToken;

import microsoft.servicefabric.services.communication.runtime.ServiceInstanceListener;
import microsoft.servicefabric.services.runtime.StatelessService;

public class CalculatorService extends StatelessService {

    @Override
    protected List<ServiceInstanceListener> createServiceInstanceListeners() {
        ArrayList<ServiceInstanceListener> listeners = new ArrayList();
        listeners.add(new ServiceInstanceListener((context) -> {
        	return new WebCommunicationListener(context);
        }));
        return listeners;
    }
}
